src/ucp/dt/dt.c

```diff
@@ -142,3 +142,43 @@ size_t ucp_dt_pack(ucp_worker_h worker, ucp_datatype_t datatype,
     state->offset += result_len;
     return result_len;
 }
+
+ucs_status_t ucp_dt_query(ucp_datatype_t datatype, ucp_datatype_attr_t *attr)
+{
+    ucp_dt_generic_t *dt_gen;
+    void             *state_gen;
+
+    /* Currently, the only datatype attribute to query is the packed size. */
+    if (!(attr->field_mask & UCP_DATATYPE_ATTR_FIELD_PACKED_SIZE)) {
+        return UCS_OK;
+    }
+
+    switch (datatype & UCP_DATATYPE_CLASS_MASK) {
+    case UCP_DATATYPE_CONTIG:
+        attr->packed_size = ucp_contig_dt_elem_size(datatype);
+        return UCS_OK;
+    case UCP_DATATYPE_IOV:
+        if (attr->buffer == NULL ||
```

Comment:
1. ( ) around == expression
2. when accessing ANY field in "attr", it should be protected by field_mask check
see for example: https://github.com/openucx/ucx/blob/6fe27d4b7a2d3dbc7ea68aee1a8f491c30a3cf57/src/ucp/core/ucp_mm.c#L655